src/L2/resolver/AddrResolver.sol

```diff
@@ -103,6 +121,28 @@ abstract contract AddrResolver is IAddrResolver, IAddressResolver, ResolverBase
         }
     }
 
+    /// @notice Fetch the uint32 coinType from an EVM coinType.
+    ///
+    /// @dev Extract Chain ID from `coinType`.
+    ///
+    /// @param coinType The coin type.
+    ///
+    /// @return The Chain ID or 0 if non-EVM Chain.
+    function chainFromCoinType(uint256 coinType) internal pure returns (uint32) {
+        if (coinType == COIN_TYPE_ETH) return CHAIN_ID_ETH;
+        coinType ^= COIN_TYPE_DEFAULT;
+        return uint32(coinType < COIN_TYPE_DEFAULT ? coinType : 0);
+    }
+
+    /// @notice Determine if `coinType` is for an EVM address.
+    ///
+    /// @param coinType The network-as-coin type.
+    ///
+    /// @return `true` if coin type represents an EVM address, else `false`.
```

Comment:
a little thrash here and above on using coin type v.s. coinType v.s. `coinType` - would be nice to standardize